docs/docs/spark-configuration.md

```diff
@@ -145,6 +145,61 @@ Using those SQL commands requires adding Iceberg extensions to your Spark enviro
+| spark.sql.iceberg.check-nullability                    | true                                                           | Whether to perform the nullability check during writes                 |
+| spark.sql.iceberg.check-ordering                       | true                                                           | Whether to check the order of fields during writes                     |
+| spark.sql.iceberg.planning.preserve-data-grouping      | false                                                          | Whether to preserve the existing grouping of data while planning splits |
+| spark.sql.iceberg.aggregate-push-down.enabled          | true                                                           | Enables pushdown of aggregate functions (MAX, MIN, COUNT)              |
+| spark.sql.iceberg.distribution-mode                    | See [Spark Writes](spark-writes.md#writing-distribution-modes) | Controls distribution strategy during writes                           |
+| spark.wap.id                                           | null                                                           | [Write-Audit-Publish](branching.md#audit-branch) snapshot staging ID   |
+| spark.wap.branch                                       | null                                                           | WAP branch name for snapshot commit                                    |
+| spark.sql.iceberg.compression-codec                    | Table default                                                  | Write compression codec (e.g., `zstd`, `snappy`)                       |
+| spark.sql.iceberg.compression-level                    | Table default                                                  | Compression level for Parquet/Avro                                     |
+| spark.sql.iceberg.compression-strategy                 | Table default                                                  | Compression strategy (for ORC)                                         |
+| spark.sql.iceberg.data-planning-mode                   | Table default                                                  | Override for data planning mode                                        |
+| spark.sql.iceberg.delete-planning-mode                 | Table default                                                  | Override for delete planning mode                                      |
+| spark.sql.iceberg.advisory-partition-size              | Table default                                                  | Advisory size (bytes) for data planning                                |
+| spark.sql.iceberg.locality.enabled                     | false                                                          | Report locality information for task planning                          |
+| spark.sql.iceberg.executor-cache.enabled               | true                                                           | Enables cache for executor-side metadata                               |
```

Comment:
I believe we only use this for Delete File Caching at the moment (no metadata)

Reply:
I will check and update accordingly

Reply:
it is indeed used only for Delete File Caching ([here]()https://github.com/apache/iceberg/blob/242717c8c516dcc49bc4368b77cd3f3af40720c8/spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java#L232).
But the config might be used in the future somewhere else and the documentation might become stale if we specifically say it is for delete file.
Should we have something more generic like:
```
Enables cache for executor-side
```

Reply:
I changed it to:
```
Enables cache for executor-side (currently used to cache Delete Files)     
```